packages/components/src/custom-select-control/test/index.js

```diff
@@ -4,54 +4,173 @@
 import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import CustomSelectControl from '..';
 
-describe( 'CustomSelectControl', () => {
-	it( 'Captures the keypress event and does not let it propagate', async () => {
-		const user = userEvent.setup();
-		const onKeyDown = jest.fn();
-		const options = [
-			{
-				key: 'one',
-				name: 'Option one',
-			},
-			{
-				key: 'two',
-				name: 'Option two',
-			},
-			{
-				key: 'three',
-				name: 'Option three',
+const props = {
+	options: [
+		{
+			key: 'flower1',
+			name: 'violets',
+		},
+		{
+			key: 'flower2',
+			name: 'crimson clover',
+		},
+		{
+			key: 'flower3',
+			name: 'poppy',
+		},
+		{
+			key: 'color1',
+			name: 'amber',
+			className: 'amber-skies',
+		},
+		{
+			key: 'color2',
+			name: 'aquamarine',
+			style: {
+				backgroundColor: 'rgb(127, 255, 212)',
+				rotate: '13deg',
 			},
-		];
+		},
+	],
+	__nextUnconstrainedWidth: true,
+};
 
-		render(
-			<div
-				// This role="none" is required to prevent an eslint warning about accessibility.
-				role="none"
-				onKeyDown={ onKeyDown }
-			>
-				<CustomSelectControl
-					options={ options }
-					__nextUnconstrainedWidth
-				/>
-			</div>
+const ControlledCustomSelectControl = ( { options } ) => {
+	const [ value, setValue ] = useState( options[ 0 ] );
+	return (
+		<CustomSelectControl
+			{ ...props }
+			onChange={ ( { selectedItem } ) => setValue( selectedItem ) }
+			value={ options.find( ( option ) => option.key === value.key ) }
+		/>
+	);
+};
+
+describe.each( [
+	[ 'uncontrolled', CustomSelectControl ],
+	[ 'controlled', ControlledCustomSelectControl ],
+] )( 'CustomSelectControl %s', ( ...modeAndComponent ) => {
+	const [ , Component ] = modeAndComponent;
+
+	it( 'Should replace the initial selection when a new item is selected', async () => {
+		const user = userEvent.setup();
+
+		render( <Component { ...props } /> );
+
+		const currentSelectedItem = screen.getByRole( 'button' );
+
+		expect( currentSelectedItem ).toHaveTextContent( 'violets' );
```

Comment:
Noting that if we reorder the fixture above, the test will break. Could make sense to access `props.options[ 0 ].name` here

Reply:
What do you think of specifying the text instead? 
```js
screen.getByRole( 'button', {
	text: 'violets',
} )
```

Reply:
Then we could check for `toBeVisible` instead? 

```suggestion
expect( currentSelectedItem ).toBeVisible();	
```

Reply:
Either way sounds good to me. Leaving it to your judgment, as it's a minor detail.